transport/conn.go

```diff
@@ -172,11 +174,61 @@ type Conn struct {
 	stream StreamIDAllocator
 }
 
+type ConnConfig struct {
+	tcpNoDelay bool
+	timeout    time.Duration
+	// This will be used.
+	// defaultConsistency frame.Consistency
+}
+
 const (
 	requestChanSize = 1024
 	ioBufferSize    = 8192
 )
 
+// OpenShardConn opens connection mapped to a specific shard on scylla node.
+func OpenShardConn(rAddr string, si ShardInfo, cfg ConnConfig) (*Conn, error) { // nolint:unused // This will be used.
```

Comment:
I'd suggest and naming scheme change

- `rAddr` to `addr`
- `lAddr` to `localAddr`